OSDOCS-3501: add clb timeouts and restructure (#45296)
openshift/openshift-docs: modules/nw-configuring-clb-timeouts.adoc

// Modules included in the following assemblies:
//
// * networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws.adoc

:_content-type: PROCEDURE
[id="nw-configuring-clb-timeouts_{context}"]
= Configuring Classic Load Balancer timeouts

> You can configure the default timeouts for a Classic Load Balancer (CLB) to extend idle connections.

Reviewer: Is a CLB an API object or something else?

Reviewer: Also, is it "AWS Classic Load Balancer" or just "Classic Load Balancer" when introduced in a module?

Author: it can be either, Classic Load Balancers are AWS-specific I was just trying to be consistent with the NLB content.

.Prerequisites

> * You need a deployed Ingress Controller on a running cluster.

Reviewer: ```suggestion
* You must have a deployed Ingress Controller on a running cluster.
```

.Procedure

> * Using the `oc patch` command, set an AWS connection idle timeout of five minutes for the default `ingresscontroller`:

Reviewer: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

Rephrasing option:

> If a step in a procedure is to run a command, make sure that the step text includes an explicit instruction to "run" or "enter" the command. In most cases, use one of the following patterns to introduce the code block:
>
>  -  \<Step description>, run the following command:
>
>  -  \<Step description>, enter the following command:

+
[source,terminal]
----
$ oc -n openshift-ingress-operator patch ingresscontroller/default \
    --type=merge --patch='{"spec":{"endpointPublishingStrategy": \
    {"type":"LoadBalancerService", "loadBalancer": \
    {"scope":"External", "providerParameters":{"type":"AWS", "aws": \
    {"type":"Classic", "classicLoadBalancer": \
    {"connectionIdleTimeout":"5m"}}}}}}}'
----
+

> To restore the default value of the timeout, run the following command:

Reviewer: Could this be a discrete, optional step?

+
[source,terminal]
----
$ oc -n openshift-ingress-operator patch ingresscontroller/default \
    --type=merge --patch='{"spec":{"endpointPublishingStrategy": \
    {"loadBalancer":{"providerParameters":{"aws":{"classicLoadBalancer": \
    {"connectionIdleTimeout":null}}}}}}}'
----

[NOTE]
====

> You must specify the `scope` field when you change the connection timeout value unless the current scope has already been set. When you set the `scope` field, you do not need to do so again when restoring the default timeout value.

Reviewer: ```suggestion
You must specify the `scope` field when you change the connection timeout value unless the current scope is already set. When you set the `scope` field, you do not need to do so again when [after?] you restore the default timeout value.
```

Author: Thanks for pointing this out, I see the confusion this sentence creates
